### ISSUE TITLE
How to wrap struct with int[]

### ISSUE THREAD

**Issue by hanslovsky** [bytedeco/javacpp#590]
I am trying to create Java bindings for the [concorde traveling salesman problem solver](https://www.math.uwaterloo.ca/tsp/concorde.html). I have had some success using JNI but would like to explore javacpp as another option that is easier to manage. I struggle to wrap a struct
```c
struct CCrandstate {
    int a;
    int b;
    int arr[55];
}
```
with `int[]` member in JavaCPP. Following #298, this is my Kotlin code (reduced to the relevant parts):
```kotlin
import org.bytedeco.javacpp.IntPointer
import org.bytedeco.javacpp.Loader
import org.bytedeco.javacpp.Pointer
import org.bytedeco.javacpp.annotation.Cast
import org.bytedeco.javacpp.annotation.MemberGetter
import org.bytedeco.javacpp.annotation.Platform
import org.bytedeco.javacpp.tools.InfoMap
import org.bytedeco.javacpp.tools.InfoMapper


@Platform(include=["concorde.h"], link=["concorde"])
class TSPSolverJavaCPP(
	val n: Int,
	val edgeWeights: Map<Pair<Int, Int>, Int>
): InfoMapper {

	override fun map(infoMap: InfoMap?) {}


	class CCrandstate private constructor(p: Pointer?) : Pointer(p) {

		constructor(): this(null) {
			allocate()
		}

		constructor(size: Long): this(null) {
			allocateArray(size)
		}

		// struct CCrandstate {
		//    int a;
		//    int b;
		//    int arr[55];
		// }

		@Cast("int")
		external fun a(): Int
		external fun a(setter: Int): CCrandstate

		@Cast("int")
		external fun b(): Int
		external fun b(setter: Int): CCrandstate

		@MemberGetter
		external fun arr(): IntPointer

		private external fun allocate()
		private external fun allocateArray(size: Long)
	}
}

fun main() {
	val rstate = TSPSolverJavaCPP.CCrandstate()
	println(rstate)
}
```
I use gradle with the `gradle-javacpp-build` plugin:
```gradle
plugins {
    // Apply the org.jetbrains.kotlin.jvm Plugin to add support for Kotlin.
    id("org.jetbrains.kotlin.jvm") version "1.5.31"
    id("org.bytedeco.gradle-javacpp-build") version "1.5.7"

    // Apply the java-library plugin for API and implementation separation.
    `java-library`
}
```
When I run
```
./gradlew :lib:javacppBuildParser
```
I see this error:
```
> Task :lib:javacppBuildParser FAILED
Could not load platform properties for class me.hanslovsky.konkorde.TSPSolver
Could not load platform properties for class me.hanslovsky.konkorde.TSPSolverJNIKt
Could not load platform properties for class me.hanslovsky.konkorde.TSPSolverJavaCPPKt
In file included from /home/zottel/workspace/jsprit-sort/sektion-sort/lib/src/main/java/jniTSPSolverJavaCPP.cpp:99:
/usr/include/concorde.h:2959:61: error: expected ‘,’ or ‘...’ before ‘new’
 2959 |     CCtsp_copy_lpcut_in (CCtsp_lpcut_in *c, CCtsp_lpcut_in *new),
      |                                                             ^~~
/usr/include/concorde.h:2972:25: error: expected ‘,’ or ‘...’ before ‘new’
 2972 |         CCtsp_lpcut_in *new),
      |                         ^~~
/usr/include/concorde.h:2973:61: error: expected ‘,’ or ‘...’ before ‘new’
 2973 |     CCtsp_copy_lpclique (CCtsp_lpclique *c, CCtsp_lpclique *new),
      |                                                             ^~~
/usr/include/concorde.h:2974:61: error: expected ‘,’ or ‘...’ before ‘new’
 2974 |     CCtsp_copy_lpdomino (CCtsp_lpdomino *c, CCtsp_lpdomino *new),
      |                                                             ^~~
/usr/include/concorde.h:3177:22: error: expected ‘,’ or ‘...’ before ‘new’
 3177 |         CCtsp_lpcut *new),
      |                      ^~~
/usr/include/concorde.h:3179:22: error: expected ‘,’ or ‘...’ before ‘new’
 3179 |         CCtsp_lpcut *new),
      |                      ^~~
/usr/include/concorde.h:3455:63: error: expected ‘,’ or ‘...’ before ‘new’
 3455 |     CCtsp_copy_skeleton (CCtsp_skeleton *old, CCtsp_skeleton *new),
      |                                                               ^~~
/usr/include/concorde.h:5847:28: error: expected ‘,’ or ‘...’ before ‘class’
 5847 |         CCverify_cutclass *class);
      |                            ^~~~~
/usr/include/concorde.h:5850:47: error: expected ‘,’ or ‘...’ before ‘class’
 5850 |     CCverify_initcutclass (CCverify_cutclass *class),
      |                                               ^~~~~
/usr/include/concorde.h:5851:47: error: expected ‘,’ or ‘...’ before ‘class’
 5851 |     CCverify_freecutclass (CCverify_cutclass *class);
      |                                               ^~~~~
/home/zottel/workspace/jsprit-sort/sektion-sort/lib/src/main/java/jniTSPSolverJavaCPP.cpp: In function ‘_jobject* Java_me_hanslovsky_konkorde_TSPSolverJavaCPP_00024CCrandstate_arr__Lorg_bytedeco_javacpp_IntPointer_2(JNIEnv*, jobject, jobject)’:
/home/zottel/workspace/jsprit-sort/sektion-sort/lib/src/main/java/jniTSPSolverJavaCPP.cpp:567:14: error: incompatible types in assignment of ‘int*’ to ‘int [55]’
  567 |     ptr->arr = (int*)ptr0;
      |     ~~~~~~~~~^~~~~~~~~~~~

FAILURE: Build failed with an exception.
```

I don't know what I am doing wrong here and how my example case is different from #298. What am I doing wrong here? I appreciate any guidance.

**Comment by saudet** [comment]
Using the Parser as explained, for example, at https://github.com/bytedeco/javacpp/wiki/Mapping-Recipes, we get the following, which works fine, but the equivalent in Kotlin would also work:
```java
public class CCrandstate extends Pointer {
    static { Loader.load(); }
    /** Default native constructor. */
    public CCrandstate() { super((Pointer)null); allocate(); }
    /** Native array allocator. Access with {@link Pointer#position(long)}. */
    public CCrandstate(long size) { super((Pointer)null); allocateArray(size); }
    /** Pointer cast constructor. Invokes {@link Pointer#Pointer(Pointer)}. */
    public CCrandstate(Pointer p) { super(p); }
    private native void allocate();
    private native void allocateArray(long size);
    @Override public CCrandstate position(long position) {
        return (CCrandstate)super.position(position);
    }
    @Override public CCrandstate getPointer(long i) {
        return new CCrandstate((Pointer)this).offsetAddress(i);
    }

    public native int a(); public native CCrandstate a(int setter);
    public native int b(); public native CCrandstate b(int setter);
    public native int arr(int i); public native CCrandstate arr(int i, int setter);
    @MemberGetter public native IntPointer arr();
}
```

**Comment by hanslovsky** [comment]
Thank you for pointing me in the right direction @saudet, this was very helpful! There were a few things that I did wrong:
 1. I only ran
     ```
     ./gradlew :lib:javacppBuildParser
     ```
     between my tests. Instead, run 
     ```
     ./gradlew :lib:clean :lib:compileKotlin :lib:compileJava :lib:javacppBuildParser
     ```
     to ensure that everything gets cleaned, built, and re-compiled.
 2. I include a C header, use `cinclude` instead of `include` in `@Platform` annotation
 3. The C header I include uses C++ keywords as identifiers (`new`, `class`). I had to rename those identifiers. After changing this I was able to compile both @saudet's Java example and my Kotlin minimum working example from above. I still had to add `Loader.load()` per @saudet's suggestion to get my Kotlin example to run.

Note: Kotlin does not allow to override `Pointer.position(long)` and `Pointer.getPointer(long)` to return `CCrandstate`. It has to be parameterized by `<P: Pointer>` to conform with super implementation. I figure Kotlin is stricter with respect to generics and return type than Java. I am not sure that I need these anyway because I will likely not use arrays of `CCrandstate`.
